org.eclipse.debug.ui/ui/org/eclipse/debug/ui/actions/ContextualLaunchAction.java

```diff
@@ -314,6 +311,9 @@ private void populateMenuItem(String mode, LaunchShortcutExtension ext, Menu men
 			label.append(accelerator);
 			label.append(' ');
 		}
+		if (indent != null) {
+			label.append(indent);
```

Comment:
if in case of "no indent" a caller passes empty string (since this is private method, you control all the call sites0 then 
label.append(indent);
can be executed unconditionally without surrounding "if"

Reply:
such micro-optimization -whose actual benefits are not obvious- comes at the risk that someone later may run in an NPE if the code is copied or made package-visible or whatever. it's safer and very cheap to handle null gracefully. It's the fundamental issue with Java that despite everything one would write in doc or whatever, `null` can always be used, better get the code handle it.

Reply:
> people write unit tests for new code, aren't' they ?

Sometimes yes, usually no.

Reply:
people write unit tests for new code, aren't' they ?

Sometimes yes, usually no.

then we shouldn't call our profession "software EGINEERING" :((((